align/pnr/checker.py

```diff
@@ -0,0 +1,26 @@
+from ..schema import constraint
+from ..cell_fabric import transformation
+
+
+def check_placement(placement_verilog_d):
+    for module in placement_verilog_d['modules']:
+        if 'constraints' not in module or len(module['constraints']) == 0:
+            continue  # No constraints
+        constraints = constraint.ConstraintDB(module['constraints'])
+        if sum(hasattr(x, 'check') for x in constraints) == 0:
+            continue  # Nothing useful to check against
+        for inst in module['instances']:
+            t = inst['transformation']
+            r = next((x['bbox'] for x in placement_verilog_d['modules'] if x['name'] == inst['template_name']),
```

Comment:
I have no idea what this does. Is there another way to write it? It is probably also has quadratic runtime. You should build a table first with the bboxes.